src/parse.rs

```diff
@@ -1846,18 +1856,58 @@ where
                 Some('(') => {
                     if self.try_consume_str("?")
                         && let Some(name) = self.try_consume_named_capture_group_name()
-                        && self
-                            .named_group_indices
-                            .insert(name, self.group_count_max)
-                            .is_some()
                     {
-                        return error("Duplicate capture group name");
+                        // Build current alternative path (depth, alt_index) pairs
+                        let mut current_path = Vec::new();
+                        for d in 0..=paren_depth {
+                            current_path.push((d, *alt_indices.get(&d).unwrap_or(&0)));
+                        }
+
+                        // Check if this name already exists in a conflicting alternative path.
+                        // Two paths conflict if they share the same alternative indices
+                        // at all common depth levels (i.e., one is a prefix of the other
+                        // or they're identical up to the shallower depth).
+                        if let Some(existing_paths) = named_groups.get(&name) {
```

Comment:
I think it would be simpler to construct the set of NamedCaptureGroups up-front and then look for conflicts in a separate path, rather than intermingling them. That is, this function should return some data structure which can then be analyzed for conflicts.